### PR TITLE
Service account: Update service accounts creation

### PR DESCRIPTION
**Pull request by vtorosyan**
### Context

The create API for service accounts supported only service account name, update API supports both to provide the isDisabled flag and also the basic role.

While in the UI we don't allow to create disabled service accounts (which probably makes sense), from the API supported the functionality would make it easier for usage and provisioning (e.g. for terraform to control the state of the resource easier). Additionally, creating service account without role programmatically means that users need to call an additional API.

### Scope of the PR

The PR updates the create request to include two new fields which allow to specify if the service account is enabled or not, and also the basic role of service account.

### To-Do
- [x] Update docs
- [x] Fix/add tests

### CODE REVIEW

**Comment by Jguer** [comment]
@vtorosyan while it is not relevant for the UI, it does respect what a CRUD resource should do. I think we should add tests/docs and merge it

**Review by Jguer — CHANGES_REQUESTED** [review]
:+1: only change I'd really consider is the transaction one.

This PR will conflict with https://github.com/grafana/grafana/pull/51819 by the way

**Comment by vtorosyan** [comment]
Thanks @Jguer for taking a look at this, will address comments and wait until your PR is merged to resolve conflicts afterwards. The important I needed to know if the extending API is OK 👍

File: pkg/services/serviceaccounts/models.go
Comment (vtorosyan) [reply pkg/services/serviceaccounts/models.go:30]: I get the duplication, but I'd like to keep them separate for now. One main reason being the `name` field, which is optional for update but required for create. This is one instance, but keeping them separate would make our life easier to evolve them in the future with minimum dependencies.

File: pkg/services/serviceaccounts/database/database.go
Comment (vtorosyan) [reply pkg/services/serviceaccounts/database/database.go]: https://github.com/grafana/grafana/pull/51848/commits/c265d562f44f52557941afd0b1b4aafca46eb6d3

**Review by Jguer — APPROVED** [review]

**Pull request merged**
